projects/com.oracle.truffle.llvm.parser.bc/src/com/oracle/truffle/llvm/parser/bc/LLVMBitcodeVisitor.java

```diff
@@ -348,10 +359,74 @@ private LLVMExpressionNode getGlobalVariable(GlobalValueSymbol global) {
             final LLVMExpressionNode loadedFunction = factoryFacade.createLoad(this, functionType, functionLoadTarget);
             final LLVMExpressionNode[] argNodes = new LLVMExpressionNode[]{factoryFacade.createFrameRead(this, LLVMBaseType.ADDRESS, stackSlot)};
             final LLVMNode functionCall = factoryFacade.createFunctionCall(this, loadedFunction, argNodes, LLVMBaseType.VOID);
-            structors[i] = functionCall;
+
+            final StructureConstant structureConstant = (StructureConstant) arrayConstant.getElement(i);
+            final Integer priorityConstant = LLVMSymbolResolver.evaluateIntegerConstant(structureConstant.getElement(0));
+            assert priorityConstant != null;
+            sortedStructors.add(priorityConstant, functionCall);
+        }
+
+        return sortedStructors.getAll();
+    }
+
+    private static final class SortedLLVMNodeList {
```

Comment:
Arrays.sort() also does not support sorting keys separate from data values, so we would not get rid of the additional element class anyways. The list just seemed a bit more expressive than a solution with Arrays.sort(). Nonetheless, if you think its preferable I can do that too. I'll replace it tomorrow.